File: leetcode/all/typescript/1447.最简分数.ts
/*
 * @lc app=leetcode.cn id=1447 lang=typescript
 *
 * [1447] 最简分数
 */

// @lc code=start

const gcd = (a, b) => (b === 0 ? a : gcd(b, a % b))

function simplifiedFractions(n: number): string[] {
    const ans = []
    for (let deno = 2; deno <= n; deno++)
        for (let mole = 1; mole < deno; mole++)
            if (gcd(mole, deno) === 1) ans.push(`${mole}/${deno}`)
    return ans
}
// @lc code=end

File: leetcode/all/python3/1400-1499/1447.最简分数.py
#
# @lc app=leetcode.cn id=1447 lang=python3
#
# [1447] 最简分数
#

# @lc code=start
from math import gcd
from typing import List


class Solution:
    def simplifiedFractions(self, n: int) -> List[str]:
        return [
            f'{j}/{i}' for i in range(2, n + 1) for j in range(1, i) if gcd(j, i) == 1
        ]


# @lc code=end
